dashboard/test/models/course_offering_test.rb

```diff
@@ -141,6 +141,32 @@ class CourseOfferingTest < ActiveSupport::TestCase
     end
   end
 
+  test "throws exception if removing course version of script that prevent course version change" do
+    script = create :script, is_course: true, family_name: 'family', version_year: '2000'
+    CourseOffering.add_course_offering(script)
+
+    script.family_name = nil
+    script.save!
+    script.reload
+    script.stubs(:prevent_course_version_change?).returns(true)
```

Comment:
nit: it's not good form to stub the object under test: https://thoughtbot.com/blog/don-t-stub-the-system-under-test what do you think about adding a resource to the script here instead?

Reply:
Interesting -- I figured as I'm not testing `prevent_course_version_change?` I could stub it. Happy to change, here and above.

Reply:
sorry, yeah, it seems like a good practice to follow, but it also seems like stubbing methods on the object under test is probably going to need to remain in our playbook when it is not super easy to avoid. 

the article recommended using this as a prompt to split the class up, but in this case we can't exactly split the model file in two, so I'm not sure exactly what the solution would be. extract some of the logic to dashboard/lib/services, maybe? I'd love to head in that direction, but not sure this is the right occasion to add that as a best practice.

anyway, I'd be happy to leave this as a "best effort" (least effort?) goal for now.